# 1.Carry out descriptive analysis on the dataset

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
data=pd.read_csv('Week8dataset.csv')

In [5]:
data.head()

Gender Test preparation  Total Marks
0    male             none           14
1  female             none           28
2  female             none           18
3  female             none           48
4  female             none           21

In [6]:
df=pd.DataFrame(data)

In [7]:
df

Gender Test preparation  Total Marks
0     male             none           14
1   female             none           28
2   female             none           18
3   female             none           48
4   female             none           21
5   female        completed           40
6     male             none           30
7   female             none           12
8     male             none           18
9     male             none           24
10    male        completed           44
11    male             none           23
12    male        completed           43
13    male             none           28
14  female             none           41
15  female             none           13
16  female        completed           48
17  female             none           45
18    male             none           47
19    male             none           36
20    male             none           14
21    male        completed           44
22  female             none           41
23  female             none           27
24  female             none           27
25  female             none           38
26  female        completed           50
27  female        completed           43

In [8]:
df.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

The given data set has 28 entries with a mean of 32.32 and a standard deviation of 12.45

# 2.Check whether male and female students have the same marks

In [9]:
df.groupby('Gender').mean()

Total Marks
Gender             
female    33.750000
male      30.416667

Hypothesis:

* H0 : The average score of female and male students is equal
* Ha : The average scores are not equal
* Alpha: 0.05
* Performing two-tailed t-test on the dataset to analyse our hypothesis

In [10]:
male=df.query('Gender=="male"')['Total Marks']
female=df.query('Gender=="female"')['Total Marks']

In [12]:
from scipy import stats

In [13]:
t_value, p_value = stats.ttest_ind(male,female, alternative='two-sided')

In [14]:
print("The t value of means is: ", round(t_value, 3))
print("The p value based on the t score is: ", round(p_value, 3))

The t value of means is:  -0.694
The p value based on the t score is:  0.494


As p value is not less than the 0.05 we cannot reject null hypothesis and we conclude that female and male students secured the same marks

# 3.Whether test preparation helps the students

Hypothesis:
 * H0: The average score of students who completed vs students who didn't is equal
 * Ha : The average scores of students who completed are higher
* Significance value  : 0.05
* Performing one-tailed t-test

In [12]:
print(df1.var())

Total Marks    155.115079
dtype: float64


In [13]:
df1.groupby('Test preparation').var()

Total Marks
Test preparation             
completed           11.285714
none               135.990476

In [14]:
df["Test preparation"].value_counts()

none         21
completed     7
Name: Test preparation, dtype: int64

In [15]:
from statsmodels.stats.weightstats import ttest_ind



In [16]:
completed=df[df['Test preparation']=="completed"]['Total Marks']
none=df[df['Test preparation']=="none"]['Total Marks']

In [20]:
t_value1, p_value1 = stats.ttest_ind(completed,none, alternative='greater')

In [22]:
print("The t value of means is: ", round(t_value1, 5))
print("The p value based on the t score is: ", round(p_value1, 5))

The t value of means is:  3.61436
The p value based on the t score is:  0.00063



As P value is much lower than  = 0.05, we reject the null hypothesis that the scores are equal and conclude that
the students who completed the course have higher average score